overlay/tun_linux.go

```diff
@@ -147,16 +150,58 @@ func newTunGeneric(c *config.C, l *logrus.Logger, file *os.File, cidr netip.Pref
 	return t, nil
 }
 
+func (t *tun) getRoutingTableID() (int, error) {
+	if t.alternativeRoutingTable == "" {
+		return unix.RT_TABLE_MAIN, nil
+	}
+
+	// Try parsing as a number first
+	if tableID, err := strconv.Atoi(t.alternativeRoutingTable); err == nil {
+		return tableID, nil
+	}
+
+	// If it's not a number, look up the table name in /etc/iproute2/rt_tables
+	content, err := os.ReadFile("/etc/iproute2/rt_tables")
```

Comment:
The location of this file seems like it could be distro-specific. Should it be configurable? Can the information instead be obtained via netlink or /proc?